picard/util/__init__.py

```diff
@@ -88,6 +88,29 @@ def unlock(self):
         self.__lock.unlock()
 
 
+class EventProcessingIterator:
+    """
+    Wrapper around an existing iterator, that calls QCoreApplication.processEvents()
+    in certain iteration steps.
+    """
```

Comment:
I think you need a note here that this enumerator should only be used in code that runs on the main thread and not on code which runs in a worker thread.

Reply:
Good point, addressed this in https://github.com/metabrainz/picard/pull/1654/commits/4dbe09445cabde0007867f166a8a5b7a31c6779e . Also rewrote the iterator using a generator.